Doc/library/pathlib.rst

```diff
@@ -946,6 +946,111 @@ call fails (for example because the path doesn't exist).
+   system. If a file or a directory is removed from or added to the *dirpath*
+   during the generation of *dirnames* and *filenames*, it is uncertain whether
+   the new entry will appear in the generated lists.
+
+   If optional argument *top_down* is ``True`` or not specified, the triple for a
+   directory is generated before the triples for any of its subdirectories
+   (directories are generated top-down).  If *top_down* is ``False``, the triple
+   for a directory is generated after the triples for all of its subdirectories
+   (directories are generated bottom-up). No matter the value of *top_down*, the
+   list of subdirectories is retrieved before the tuples for the directory and
+   its subdirectories are generated.
+
+   When *top_down* is True, the caller can modify the *dirnames* list in-place
+   (For example, using :keyword:`del` or slice assignment), and :meth:`Path.walk`
+   will only recurse into the subdirectories whose names remain in *dirnames*;
+   this can be used to prune the search, or to impose a specific order of visiting,
+   or even to inform :meth:`Path.walk` about directories the caller creates or
+   renames before it resumes :meth:`Path.walk` again. Modifying *dirnames* when
+   *top_down* is False has no effect on the behavior of :meth:`Path.walk()`, since the
+   directories in *dirnames* have already been generated by the time *dirnames*
+   is yielded to the caller.
+
+   By default errors from :meth:`Path._scandir` call are ignored.  If
+   optional argument *on_error* is specified, it should be a callable; it
+   will be called with one argument, an :exc:`OSError` instance. It can
+   report the error to continue with the walk, or raise the exception
+   to abort the walk. Note that the filename is available as the
+   ``filename`` attribute of the exception object.
+
+   By default, :meth:`Path.walk` will not walk down into symbolic links that
+   resolve to directories.  Set *follow_symlinks* to ``True`` to visit directories
+   pointed to by symlinks, on systems that support them.
+
+   .. note::
+
+      Be aware that setting *follow_symlinks* to ``True`` can lead to infinite
+      recursion if a link points to a parent directory of itself. :meth:`Path.walk`
+      does not keep track of the directories it visited already.
+
+   .. note::
+      If self is a relative Path, don't change the current working directory between
+      resumptions of :meth:`Path.walk`. :meth:`Path.walk` never changes the current
+      directory, and assumes that the caller doesn't either.
+   .. note::
+
+      :meth:`Path.walk` assumes the directories have not been modified between
+      its resumptions. I.e. If a directory from *dirnames* has been replaced
+      with a symlink and *follow_symlinks* = ``False``, :meth:`Path.walk` will
+      still try to descend into it. To prevent such behavior, remove directories
+      from *dirnames* if they have been modified and you do not want to
+      descend into them anymore.
```

Comment:
This note feels like unnecessary detail to me - replacing a directory with a symlink is an edge case, and it can already be inferred from the prior section on dirnames that entries should be removed from *dirnames* if they no longer exist, or are no longer directories.

Reply:
I feel like having it at the end of the documentation of the method won't hurt.
1) The developers of os.walk decided that it made sense to put it in
2) It can resolve some rare confusion from users that don't know how scandir works and expect that the method is generating new stuff with every new iteration.